packages/block-library/src/embed/core-embeds.js

```diff
@@ -338,4 +339,18 @@ export const others = [
 		},
 		patterns: [ /^https?:\/\/wordpress\.tv\/.+/i ],
 	},
+	{
+		name: 'core-embed/amazon-kindle',
+		settings: {
+			title: 'Amazon Kindle',
+			icon: embedAmazonIcon,
+			keywords: [ __( 'ebook' ) ],
+			responsive: false,
+			description: __( 'Embed Amazon Kindle content.' ),
+		},
+		patterns: [
+			/^https?:\/\/([a-z0-9-]+\.)?(amazon|amzn)(\.[a-z]{2,4})+\/.+/i,
+			/^https?:\/\/(www\.)?(a\.co|z\.cn)\/.+/i,
+		],
```

Comment:
@Soean: This is too eagerly catching items that aren't books and attempting to render them as such.

Try pasting any product URL matching `https://www.amazon.com/gp/product/<product code>/` or `https://www.amazon.com/-/<locale>/dp/<product code>/`, for instance.

## Example

Pasting URL `https://www.amazon.com/dp/B0198CGQEE/` yields:

![output](https://user-images.githubusercontent.com/150562/101353963-548c7480-388c-11eb-95cc-adb13f94bcc4.gif)

Reply:
Reported in #27547